packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -267,18 +283,13 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
     };
   }
 
-  onSetOptions(options: OnSetOptionsProps<LinkOptions>): void {
-    if (options.changes.autoLink.changed) {
-      const [newSuggester] = this.createSuggesters();
-
-      if (options.changes.autoLink.value === true && newSuggester) {
-        this.store.addSuggester(newSuggester);
-      }
-
-      if (options.changes.autoLink.value === false) {
-        this.store.removeSuggester(this.name);
-      }
-    }
+  protected init() {
+    super.init();
+    const { autoLinkRegex } = this.options;
+    this._autoLinkRegexNonGlobal = new RegExp(
+      `^{autoLinkRegex.source}$`,
+      autoLinkRegex.flags.replace('g', ''),
+    );
```

Comment:
This is icky 🤢 , but it is either this or reseting `this.autoLinkRegex.lastIndex = 0` after each match